module/metrics/rest_api.go

```diff
@@ -0,0 +1,137 @@
+package metrics
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/slok/go-http-metrics/metrics"
+	metricsProm "github.com/slok/go-http-metrics/metrics/prometheus"
+	"github.com/slok/go-http-metrics/middleware"
+	"github.com/slok/go-http-metrics/middleware/std"
+
+	"github.com/gorilla/mux"
+)
+
+// Example recorder taken from:
+// https://github.com/slok/go-http-metrics/blob/master/metrics/prometheus/prometheus.go
+type CustomRecorder interface {
+	metrics.Recorder
+	AddTotalRequests(ctx context.Context, service string, id string)
+}
+
+type recorder struct {
+	httpRequestDurHistogram   *prometheus.HistogramVec
+	httpResponseSizeHistogram *prometheus.HistogramVec
+	httpRequestsInflight      *prometheus.GaugeVec
+	httpRequestsTotal         *prometheus.GaugeVec
+}
+
+// NewRecorder returns a new metrics recorder that implements the recorder
+// using Prometheus as the backend.
+func NewRecorder(cfg metricsProm.Config) CustomRecorder {
```

Comment:
we typically use the term `Collector` for metrics objects
```suggestion
func NewRestCollector(cfg metricsProm.Config) RestCollector {
```